Build BinaryDelta for Sparkle

mbacchi opened the pull request:
Output will go into vendor/sparkle/build
tree as opposed to src/out tree as it will
not be bundled with Brave, but rather used
to build the Sparkle delta.

Resolves: https://github.com/brave/Sparkle/issues/9

simonhong commented:
delta output also should be specified here - https://github.com/brave/Sparkle/blob/master/BUILD.gn#L22
Hmm, if output is not in src/out, we can't specify delta file in output in BUILD.gn.
Usually, all generated/compiled ouputs are in src/out/... in chromium project.
IMO, it's fine to put delta output to src/out even if it's not bundled. WDYT?
Also, we can verify the result if we specify it in output.

mbacchi commented:
@simonhong Here is an update which addresses your comments. It's correct that we couldn't add this to `outputs` if we were placing the compiled results in the `src/brave` directory, so this commit uses the `src/out` tree but does not add these files to any bundling targets. Let me know what you think.
